### ISSUE TITLE
bug about Duration or Timer ?

### ISSUE THREAD

**Issue by Farer** [dart-lang/sdk#42280]
Maybe I found a bug about **Duration** or **Timer**.

```
Timer startTimeout(int microseconds) {
  const ms = const Duration(microseconds: 1);
  var duration = ms*microseconds;
  return new Timer(duration, doSomething);
}
```

It works well If microseconds is greater than 999.
It's ok from 1000.

It doesn't works well if microseconds less than 1000.
It's not ok from 999 to below.
It responds so fast if microseconds is less than 1000.
Maybe 999 and 1 are same as far as I think.

**Comment by srawlins** [comment]
Are you perchance confusing microseconds with milliseconds? 1000 microseconds is _really_ quick.

**Comment by mraleph** [comment]
It's unclear what you are trying to achieve. In general timers and time measurements have certain granularity - if you go beyond that granularity you will see strange effects, like for example timer firing all the time.

**Comment by Farer** [comment]
@srawlins 
> Are you perchance confusing microseconds with milliseconds? 1000 microseconds is _really_ quick.

I'm not confused.


@mraleph 
> It's unclear what you are trying to achieve. In general timers and time measurements have certain granularity - if you go beyond that granularity you will see strange effects, like for example timer firing all the time.

ok.
I'll explain my plan.

There is a sprite image.
The size is 1280x1280.
It has 100 frames.
All frame images 128x128.
I want to animate these frames.
It needs to loop 100 times so quickly but also smooth animation.
So I tried to loop it quickly.

It took about one second to animate 100 frames.
When I set 1000 microseconds, it is a good.
But I wanted speed up it about 2 times.
So I set 500.
It didn't takes half time but very short time for 100 frames. ( maybe 0.1 second )
When I set 999, it is same with 500.
I think all the same when I set it from 1 to  999 microseconds.

Is there anything wrong?
What can I do?
